app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/CategoryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,82 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.spendster.R;
+import com.spendster.data.entity.Category;
+
+import java.util.List;
+
+public class CategoryRecyclerViewAdapter extends RecyclerView.Adapter<CategoryRecyclerViewAdapter.CategoryViewHolder> {
+
+    private final Context context;
+    private List<Category> categories;
+    private final OnCategoryClickListener onCategoryClickListener;
+
+    public CategoryRecyclerViewAdapter(Context context, List<Category> categories, OnCategoryClickListener onCategoryClickListener) {
```

Comment:
@SimonMaha should be a constructor without categories parameter. In case we don't have any data at the beginning.